core/vm/intpool.go

```diff
@@ -65,3 +71,54 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolPoolLimit = 5
+const (
+	PoolPoolTypeMiner = iota
+	PoolPoolTypeImporter
+	PoolPoolTypeTracer
+	PoolPoolTypeRPC
+	PoolPoolTypes
+)
+
+// IntPoolPool manages a pool of intPools
+type IntPoolPool struct {
+	pools [PoolPoolTypes][poolPoolLimit]*intPool
+
+	// Mutex
+	locks [PoolPoolTypes]sync.RWMutex
+}
+
+var intPoolPool = IntPoolPool{}
+
+// get is looking for an available pool to return
+func (ipp *IntPoolPool) get(poolType int) *intPool {
+	if poolType > PoolPoolTypes {
+		panic("Invalid intPool pool type requested")
+	}
+
+	ipp.locks[poolType].Lock()
+	defer ipp.locks[poolType].Unlock()
+
+	for i, ip := range ipp.pools[poolType] {
+		if ip.inUse == false {
+			ip.inUse = true
+			ip.poolIndex = i
+			return ip
+		}
+	}
+
+	panic("Could not find a pool")
+}
+
+// release a pool that has been allocated with Get
+func (ipp *IntPoolPool) release(ip *intPool) {
```

Comment:
@karalabe I'm really not a fan of having to manually release pools. I could come up with a general release thing (for e.g. when a transaction has been processed). Interested in your input though.

Reply:
Why? It's similar to mutex lock/unlock, to sync.pool get/out, to calling Close on almost everything.

But your mechanism is convoluted. Only use a single slice of pools, you can then fetch from there, or put back when you're done.